### PR TITLE
Add debug logs to print Request/Response data in  Databricks provider

### PR DESCRIPTION
**Pull request by rawwar**
This PR intends to add additional debug logging around Requests/Responses to/from Databricks API.

### CODE REVIEW

File: airflow/providers/databricks/hooks/databricks_base.py [hunk 1]
```diff
@@ -615,15 +623,24 @@ async def _a_do_api_call(self, endpoint_info: tuple[str, str], json: dict[str, A
...
                         url,
                         json=json,
                         auth=auth,
                         headers={**headers, **self.user_agent_header},
                         timeout=self.timeout_seconds,
                     ) as response:
+                        self.log.debug("Response Status Code: %s", response.status_code)
+                        self.log.debug("Response text: %s", response.text)
                         response.raise_for_status()
-                        return await response.json()
```
Comment (pankajkoti) [inline airflow/providers/databricks/hooks/databricks_base.py:626]: Why are we removing the await?

**Review by pankajkoti — COMMENTED** [review]
Looks good to me, one quick question inline

File: airflow/providers/databricks/hooks/databricks_base.py [hunk 2]
```diff
@@ -615,15 +623,24 @@ async def _a_do_api_call(self, endpoint_info: tuple[str, str], json: dict[str, A
...
+                        return response.json()
```
Comment (rawwar) [inline airflow/providers/databricks/hooks/databricks_base.py]:
```suggestion
                        return await response.json()
```

File: airflow/providers/databricks/hooks/databricks_base.py (hunk 1 above)
Comment (rawwar) [reply airflow/providers/databricks/hooks/databricks_base.py:626]: that was a mistake.

File: airflow/providers/databricks/hooks/databricks_base.py [hunk 3]
```diff
@@ -569,7 +576,8 @@ def _do_api_call(
                         headers=headers,
                         timeout=self.timeout_seconds,
                     )
-                    response.raise_for_status()
+                    self.log.debug("Response Status Code: %s", response.status_code)
+                    self.log.debug("Response text: %s", response.text)
```
Comment (rawwar) [inline airflow/providers/databricks/hooks/databricks_base.py:580]:
```suggestion
                    self.log.debug("Response text: %s", response.text)
                    response.raise_for_status()
```

**Review by pankajkoti — APPROVED** [review]

**Review by Lee-W — APPROVED** [review]

**Pull request merged**